source/03-components/table/table--sortable.es6.ts

```diff
@@ -0,0 +1,295 @@
+ * @return {HTMLTableCellElement[]}
+ */
+const getColumnHeaders = (table: HTMLTableElement): HTMLTableCellElement[] => {
+  const headers = Array.from(
+    table.querySelectorAll<HTMLTableCellElement>(SORTABLE_HEADER)
+  );
+  return headers.filter(header => header.closest(TABLE) === table);
+};
+
+/**
+ * Update the button label within the given header element, resetting it
+ * to the default state (ready to sort ascending) if it’s no longer sorted.
+ * @param {HTMLTableCellElement} header
+ */
+const updateSortLabel = (header: HTMLTableCellElement): void => {
+  const headerName = header.innerText;
+  const sortedAscending = header.getAttribute(SORTED) === ASCENDING;
+  const isSorted =
+    header.getAttribute(SORTED) === ASCENDING ||
+    header.getAttribute(SORTED) === DESCENDING ||
+    false;
+  const headerLabel = `${headerName}, sortable column, currently ${
+    isSorted
+      ? `${sortedAscending ? `sorted ${ASCENDING}` : `sorted ${DESCENDING}`}`
+      : 'unsorted'
+  }`;
+  const headerButtonLabel = `Click to sort by ${headerName} in ${
+    sortedAscending ? DESCENDING : ASCENDING
+  } order.`;
+  header.setAttribute('aria-label', headerLabel);
+  header.querySelector(SORT_BUTTON)?.setAttribute('title', headerButtonLabel);
+};
+
+/**
+ * Remove the aria-sort attribute on the given header element, and reset the
+ * label and button icon.
+ * @param {HTMLTableCellElement} header
+ */
+const unsetSort = (header: HTMLTableCellElement): void => {
+  header.removeAttribute(SORTED);
+  updateSortLabel(header);
+};
+
+/**
+ * Sort rows either ascending or descending, based on a given header’s
+ * aria-sort attribute.
+ * @param {HTMLTableCellElement} header
+ * @param {boolean} isAscending
+ * @return {boolean}
+ */
+const sortRows = (
+  header: HTMLTableCellElement,
+  isAscending: boolean
+): boolean => {
+  header.setAttribute(SORTED, isAscending === true ? DESCENDING : ASCENDING);
+  updateSortLabel(header);
+
+  const table = header.closest(TABLE);
+  const headerRow = header.parentNode;
+
+  if (!table || !headerRow) {
+    return false;
+  }
+
+  const tbody = table.querySelector('tbody') as HTMLTableSectionElement;
+  const siblingHeaders = headerRow.children as HTMLCollection;
+
+  Array.from(tbody.querySelectorAll('tr'))
+    .sort(
+      compareFunction(Array.from(siblingHeaders).indexOf(header), !isAscending)
+    )
+    .forEach(tr => tbody.appendChild(tr));
+
+  return true;
+};
+
+/**
+ * Update the live region immediately following the table whenever the sort
+ * changes.
+ * @param {HTMLTableElement} table
+ * @param {HTMLTableCellElement} sortedHeader
+ */
+const updateLiveRegion = (
+  table: HTMLTableElement,
+  sortedHeader: HTMLTableCellElement
+): void => {
+  const caption = table.querySelector('caption')?.innerText;
+  const sortedAscending = sortedHeader.getAttribute(SORTED) === ASCENDING;
+  const headerLabel = sortedHeader.innerText;
+  const liveRegion = table.nextElementSibling as HTMLElement;
```

Comment:
Suggest not casting and instead doing a check, as `liveRegion` could possibly not be found and be `null`:
```ts
const liveRegion = table.nextElementSibling;
if (!liveRegion || !(liveRegion instanceof HTMLElement)) {
  return;
}
```